app/services/user-language.js

```diff
@@ -0,0 +1,67 @@
+import Ember from 'ember';
+import C from 'ui/utils/constants';
+import { ajaxPromise } from 'ember-api-store/utils/ajax-promise';
+
+export default Ember.Service.extend({
+  prefs            : Ember.inject.service(),
+  session          : Ember.inject.service(),
+  intl             : Ember.inject.service(),
+  translationsList : Ember.computed.alias('app.translations'),
+  loadedLocales    : null,
+
+  bootstrap: function() {
+    this.set('loadedLocales', []);
+  }.on('init'),
+
+  initLanguage() {
+    const upl           = this.getLanguage(); // get language from user prefs
+    const uplLocal      = this.get('session').get(C.PREFS.LANGUAGE); // get local language
+    let defaultLanguage = C.LANGUAGE.DEFAULT;
+
+    if (upl) { // if user prefs
+      if (upl !== defaultLanguage) {
+        defaultLanguage = upl;
+      }
+      if (upl !== uplLocal) { // doesnt match local?
+        // set uplLocal to upl
+        this.get('session').set(C.PREFS.LANGUAGE, defaultLanguage);
+      }
+    } else { // no user pref language
+      if (uplLocal) {
+
+        defaultLanguage = uplLocal;
+
+        this.set(`prefs.${C.PREFS.LANGUAGE}`, defaultLanguage);
+      }
+    }
+
+    this.sideLoadLanguage(defaultLanguage);
+  },
+
+  getLanguage() {
+    const languagePrefs = `prefs.${C.PREFS.LANGUAGE}`;
+
+    return this.get(languagePrefs);
+  },
+
+  sideLoadLanguage(language) {
+    let loadedLocales = this.get('loadedLocales');
+    if (loadedLocales.contains(language)) {
+      this.get('intl').setLocale(language);
```

Comment:
This should probably return a promise in either case (`return Ember.RSVP.resolve()`)